src/Cryptol/Parser.y

```diff
@@ -504,24 +502,24 @@ qop                              :: { LPName }
 
 op                               :: { LPName }
   : pat_op                          { $1 }
-  | '#'                             { Located $1 $ mkUnqual $ mkInfix "#" }
-  | '@'                             { Located $1 $ mkUnqual $ mkInfix "@" }
+  | '#'                             { Located $1 $ mkUnqualUser $ mkInfix "#" }
+  | '@'                             { Located $1 $ mkUnqualUser $ mkInfix "@" }
 
 pat_op                           :: { LPName }
   : other_op                        { $1 }
 
     -- special cases for operators that are re-used elsewhere
-  | '*'                             { Located $1 $ mkUnqual $ mkInfix "*" }
-  | '+'                             { Located $1 $ mkUnqual $ mkInfix "+" }
-  | '-'                             { Located $1 $ mkUnqual $ mkInfix "-" }
-  | '~'                             { Located $1 $ mkUnqual $ mkInfix "~" }
-  | '^^'                            { Located $1 $ mkUnqual $ mkInfix "^^" }
-  | '<'                             { Located $1 $ mkUnqual $ mkInfix "<" }
-  | '>'                             { Located $1 $ mkUnqual $ mkInfix ">" }
+  | '*'                             { Located $1 $ mkUnqualUser $ mkInfix "*" }
```

Comment:
Maybe `mkUnqual` should do what `mkUnqualUser` does, and we should have `mkUnqualSystem` for the cases when we need to make a System name.  This should remove the need for all of these changes.